crates/server-primitives/src/admin.rs

```diff
@@ -5,9 +5,10 @@ use serde_json::Value;
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
 pub struct InstallApplicationRequest {
+    pub contract_app_id: String,
     pub url: url::Url,
     pub version: Option<semver::Version>,
-    pub hash: calimero_primitives::hash::Hash,
+    pub hash: Option<calimero_primitives::hash::Hash>,
```

Comment:
What is a scenario of hash missing?

Reply:
Checking if the downloaded application hash is the same as the one saved in the contract. As the source can be corrupt

Reply:
Shouldn't it then make an application invalid? And not just the hash part

Reply:
So previous implementation that I created downloaded the application wasm, check the hash and if they don't match delete the application and return message why it failed.
Can you confirm the updated logic cc @miraclx

Reply:
Take the CLI for example, `meroctl app install --url <URL>`

I know where the blob is, but I don't have the hash

For me to know the hash, I would have already downloaded it. In which case, I would use `meroctl app install --path <PATH>` which also, doesn't use the hash.

The idea of the hash is (if) you have it, provide it and we'll test the blob before using it.

And if not, you're implicitly saying you trust it regardless of what we end up downloading.